internal/apierrors/apierror.go

```diff
@@ -0,0 +1,183 @@
+package apierrors
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+)
+
+var _ error = (*StatusError)(nil)
+
+type KnownReason int
+
+const (
+	KnownReasonBadRequest KnownReason = iota
```

Comment:
I guess I forgot what I did last summer. There must have been a reason why I was using both fields in the payment service. Maybe I wanted to make sure that there is at least one fallback to validate if an error is of a certain http status, which included the status or the knownreason.

I will check this during implementation if this might still be relevant or not and will remove it accordingly